spec/unit/classes/smartd_spec.rb

```diff
@@ -1,541 +1,414 @@
 require 'spec_helper'
 
 describe 'smartd', :type => :class do
+  let(:file_header) { Regexp.new('^# Managed by Puppet -- do not edit!$') }
 
-  shared_examples_for 'default' do |values|
-    content = nil
-    if values && values[:content]
-      content = values[:content]
-    else
-      content = [
-        'DEFAULT -m root -M daily',
-        'DEVICESCAN',
-      ]
-    end
-
-    config_file = nil
-    if values && values[:config_file]
-      config_file = values[:config_file]
-    else
-      config_file = '/etc/smartd.conf'
-    end
-
-    group = nil
-    if values && values[:group]
-      group = values[:group]
-    else
-      group = 'root'
-    end
-
-    it { should contain_package('smartmontools').with_ensure('present') }
-    it do
-      should contain_file(config_file).with({
-        :ensure  => 'present',
-        :owner   => 'root',
-        :group   => group,
-        :mode    => '0644',
-      })
-    end
-    it "should contain File[#{config_file}] with correct contents" do
-      verify_contents(catalogue, config_file, content)
-    end
-  end
-
-  describe 'on a supported osfamily, default parameters' do
-    describe 'for osfamily SuSE' do
-      let(:facts) {{ :osfamily => 'SuSE', :smartmontools_version => '5.43', :gid => 'root' }}
-
-      it_behaves_like 'default', {}
-      it { should_not contain_augeas('shell_config_start_smartd') }
-      it { should contain_service('smartd').with_ensure('running').with_enable(true) }
-      it { should contain_file('/etc/smartd.conf').with_notify('Service[smartd]') }
-    end
-
-    describe 'for osfamily RedHat' do
-      describe 'for operatingsystem RedHat' do
-        describe 'for operatingsystemmajrelease 6' do
-          let(:facts) {{ :osfamily => 'RedHat', :operatingsystem => 'RedHat', :operatingsystemmajrelease => '6', :smartmontools_version => '5.43', :gid => 'root' }}
-
-          it_behaves_like 'default', {}
-          it { should_not contain_augeas('shell_config_start_smartd') }
-          it { should contain_service('smartd').with_ensure('running').with_enable(true) }
-          it { should contain_file('/etc/smartd.conf').with_notify('Service[smartd]') }
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) do
+        facts
+      end
+
+      case facts[:osfamily]
+      when 'RedHat'
+        service_name = 'smartd'
+        smartd_conf_path = case facts[:operatingsystem]
+                           when 'RedHat', 'CentOS', 'Scientific', 'OracleLinux', 'SLC', 'OEL'
```

Comment:
It does.  When a case statement is used to assign only one variable it wants the `var = case..` syntax, and it wants the keywords of the case statement to align.  I can disable the check if you want.  I'm not particularly fond of it myself, though I prefer to follow the defaults unless there's a good reason to differ.